libsolidity/interface/CompilerStack.cpp

```diff
@@ -225,6 +225,15 @@ void CompilerStack::setEVMVersion(langutil::EVMVersion _version)
 	m_evmVersion = _version;
 }
 
+void CompilerStack::setEOFVersion(std::optional<uint8_t> _version)
+{
+	if (m_stackState >= CompilationSuccessful)
+		solThrow(CompilerError, "Must set EOF version before compiling.");
+	if (_version && _version != 1)
```

Comment:
What are the possible values for the EOF version? I assume `1` is the only value supported at the moment? It would be nice to at least document this somewhere in the code. i.e. `nullopt` for legacy, `1` for EOF, and everything else is invalid. Also, are there plans to go beyond version `1`?

Reply:
Yes, the initial EOF version will be "1", so the choices for the time being will be ``nullopt`` for legacy and ``1`` for EOF.

In the future there may theoretically be further EOF versions, although we can expect 1 to be the only one mid-term (no way there will be EOF 2 within the next year).

Reply:
Is there any need to actually document this? The code is rather self-explanatory here, isn't it?
It literally says "If an eof version is requested and it's not 1, it's an invalid EOF version", right there.

Reply:
Why? The version in EOF is a single byte field, which will correspond to this value here. Why build an enum on top of that?

Reply:
I mean, technically you can define an enum to use `uint8_t`, which I think would look nicer. Does the EOF version end up in bytecode?

Reply:
Disregard the enum comment then :D